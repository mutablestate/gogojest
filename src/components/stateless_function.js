import React from 'react'

export const MyComponent = ({ countries })  => {
  const list = countries.map((country, i) => (
    <li key={i}>{country.name}</li>
  ))

  return (
    <div>
      <h1>Countries</h1>
      <ul>
        {list}
      </ul>
    </div>
  )
}
