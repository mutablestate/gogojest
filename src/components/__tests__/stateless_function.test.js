import React from 'react' // needed to use JSX
import renderer from 'react-test-renderer'

import { MyComponent } from '../stateless_function'

test('JSX list', () => {
  const countries = [{ name: 'England' }, { name: 'Thailand' }, { name: 'Hong Kong'}]
  const component = renderer.create(<MyComponent countries={countries} />)
  expect(component).toMatchSnapshot()
})
