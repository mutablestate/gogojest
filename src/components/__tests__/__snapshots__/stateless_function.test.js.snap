exports[`test JSX list 1`] = `
<div>
  <h1>
    Countries
  </h1>
  <ul>
    <li>
      England
    </li>
    <li>
      Thailand
    </li>
    <li>
      Hong Kong
    </li>
  </ul>
</div>
`;
